py-polars/tests/unit/functions/test_functions.py

```diff
@@ -183,6 +183,14 @@ def test_concat_vertical() -> None:
     assert_frame_equal(result, expected)
 
 
+def test_extend_ints() -> None:
+    a = pl.DataFrame({"a": [1 for _ in range(1)]})
+    with pytest.raises(pl.exceptions.SchemaError):
+        # This is current behavior. It would be nice to have
+        # DataFrame and lit treat ints consistently
+        a.extend(a.select(pl.lit(0).alias("a")))
```

Comment:
We should explicitly declare the dtypes in this test to future-proof it as it is dtype-sensitive - can you do:

```python
a = pl.DataFrame({"a": [1]}, schema={"a": pl.Int64})
...
a.extend(a.select(pl.lit(0, dtype=pl.Int32).alias("a")))
```

Reply:
done